java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/ShareFileController.java

```diff
@@ -0,0 +1,245 @@
+		if (shareFile == null) {
+			return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
+		}
+		boolean success = shareFileService.editAccessList(shareFile, accessMap);
+		if (!success) {
+			return ResponseEntity.badRequest().body("Error editing access list");
+		}
+		return ResponseEntity.ok("Access list edited");
+	}
+
+	@PostMapping(path = {"/update-requests"}, produces = "application/json")
+	public ResponseEntity<String> updateRequests(@RequestBody Map<Integer, String> requests,
+	                                             @RequestParam long fileId) {
+		PremiumUserDevicesRepository.PremiumUserDevice dev = userdataService.checkUser();
+		if (dev == null) {
+			return userdataService.tokenNotValidResponse();
+		}
+		ShareFileRepository.ShareFile shareFile = shareFileService.getFileById(fileId);
+		if (shareFile == null) {
+			return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
+		}
+		if (shareFile.ownerid != dev.userid) {
+			return ResponseEntity.badRequest().body("You are not the owner of this file");
+		}
+		boolean success = shareFileService.updateRequests(requests);
+		if (!success) {
+			return ResponseEntity.badRequest().body("Error updating requests");
+		}
+		shareFile = shareFileService.getFileById(fileId);
+		ShareFileRepository.ShareFileDTO shareFileDto = new ShareFileRepository.ShareFileDTO(shareFile, true);
+
+		return ResponseEntity.ok(gson.toJson(shareFileDto));
+	}
+
+	@GetMapping(path = {"/change-share-type"}, produces = "application/json")
+	public ResponseEntity<String> changeShareType(@RequestParam String filePath,
+	                                              @RequestParam String fileType,
+	                                              @RequestParam String shareType,
+	                                              @RequestParam boolean createIfNotExists) {
+		PremiumUserDevicesRepository.PremiumUserDevice dev = userdataService.checkUser();
+		if (dev == null) {
+			return userdataService.tokenNotValidResponse();
+		}
+		ShareFileRepository.ShareFile shareFile = shareFileService.getFileByOwnerAndFilepath(dev.userid, filePath);
+		if (shareFile == null) {
+			if (createIfNotExists) {
+				PremiumUserFilesRepository.UserFile userFile = userdataService.getUserFile(filePath, fileType, null, dev);
+				shareFile = shareFileService.createShareFile(userFile, false, null);
+			} else {
+				return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
+			}
+		}
+		boolean success = shareFileService.changeFileShareType(shareFile, shareType);
+		if (!success) {
+			return ResponseEntity.badRequest().body("Error changing share type");
+		}
+		if (shareType.equals(PRIVATE_SHARE_TYPE)) {
+			return ResponseEntity.ok(FILE_WAS_DELETED);
+		}
+		shareFile = shareFileService.getFileByOwnerAndFilepath(dev.userid, filePath);
+		ShareFileRepository.ShareFileDTO shareFileDto = new ShareFileRepository.ShareFileDTO(shareFile, !shareFile.isPublicAccess());
```

Comment:
!shareFile.isPublicAccess() looks slightly awkward (sounds like you toggle it on/off every time)